src/arviz_plots/plots/ppcplot.py

```diff
@@ -0,0 +1,507 @@
+
+    # observed args logic check:
+    # observed will be True/False depending on user-input/group- true for posterior, false for prior
+    # if observed and no plot_kwargs['observed'], no prob
+    # if observed = True and plot_kwargs['observed'] = True, no prob (observed is plotted)
+    # if observed = True and plot_kwargs['observed'] = False, prob (error/warning raised)
+    # if observed = False and plot_kwargs['observed'] = True, prob (error/warning raised)
+    # if observed = False and plot_kwargs['observed'] = False, no prob (observed is not plotted)
+    if (
+        observed
+        and plot_kwargs.get("observed", True) is False
+        or not observed
+        and plot_kwargs.get("observed", False) is not False
+    ):
+        # raise warning or error
+        raise ValueError(
+            f"""
+            `observed` and `plot_kwargs["observed"]` inconsistency detected.
+            `observed` = {observed}{default_observed}
+            `plot_kwargs["observed"]` = {plot_kwargs["observed"]}
+            Please make sure `observed` and `plot_kwargs["observed"]` have the same value."""
+        )
+
+    # same check for aggregate:
+    if (
+        aggregate
+        and plot_kwargs.get("aggregate", True) is False
+        or not aggregate
+        and plot_kwargs.get("aggregate", False) is not False
+    ):
+        # raise warning or error
+        raise ValueError(
+            f"""
+            `aggregate` and `plot_kwargs["aggregate"]` inconsistency detected.
+            `aggregate` = {aggregate}
+            `plot_kwargs["aggregate"]` = {plot_kwargs["aggregate"]}
+            Please make sure `aggregate` and `plot_kwargs["aggregate"]` have the same value."""
+        )
+
+    # making sure both posterior/prior predictive group and observed_data group exists in
+    # datatree provided
+    if observed:
+        for group_name in (predictive_data_group, "observed_data"):
+            if group_name not in dt.children:
+                raise TypeError(f'`data` argument must have the group "{group_name}" for ppcplot')
+    else:
+        if f"{predictive_data_group}" not in dt.children:
+            raise TypeError(f'`data` argument must have the group "{group}_predictive" for ppcplot')
+
+    # making sure kde type is one of these three
+    if kind.lower() not in ("kde", "ecdf", "scatter"):
+        raise TypeError("`kind` argument must be either `kde`, `ecdf`, or `scatter`")
+
+    if backend is None:
+        if plot_collection is None:
+            backend = rcParams["plot.backend"]
+        else:
+            backend = plot_collection.backend
+    # getting plot backend to get default colors to put in default kwargs for visual elements
+    plot_bknd = import_module(f".backend.{backend}", package="arviz_plots")
+
+    # pp distribution group plotting logic
+    pp_distribution = process_group_variables_coords(
+        dt, group=predictive_data_group, var_names=var_names, filter_vars=filter_vars, coords=coords
```

Comment:
`data_pairs` should be taken into account here:

```suggestion
        dt, group=predictive_data_group, var_names=None if var_names is None else [data_pairs.get(var_name, var_name) for var_name in var_names], filter_vars=filter_vars, coords=coords
```